### PR TITLE
NameRecordVisitor: include whole sequence of character variants' UI labels, not just the first

### LINKED ISSUE #3516: Bloated subset when using --retain-gids
Hello team.

I have a font collection file `Noto Sans CJK` which has a size of ~18MB

When I subset the font using the following comand. `pyftsubset notosanscjksc.ttf --unicodes="0-FF" --output-file="notosanscjksc-2.ttf"  --font-number=2` , it creates a ubset of size 24KB.

But when I use the arg `--retain-gids` as follows `pyftsubset notosanscjksc.ttf --unicodes="0-FF" --output-file="notosanscjksc-1.ttf"  --font-number=2 --retain-gids`, it creates a subset of size 762KB.

I have attached all 3 files here. Can you please help us understand why this flag causes the subset to be so bloated, and is there a way we can reduce the size with this flag?
[Documents.zip](https://github.com/fonttools/fonttools/files/15393489/Documents.zip)

Thanks.

### PR DESCRIPTION
**Pull request by anthrotype**
Fixes https://github.com/fonttools/fonttools/issues/3616

### CODE REVIEW

**Comment by behdad** [comment]
Lgtm.

**Comment by anthrotype** [comment]
@behdad thanks. I also added [f51a874](https://github.com/fonttools/fonttools/pull/3617/commits/f51a874611e2d067a7b9cbd62a8c0972870c5590)

thanks to Ginger for spotting

**Comment by behdad** [comment]
Ah, thanks.

File: Lib/fontTools/ttLib/tables/_n_a_m_e.py [hunk 1]
```diff
@@ -1197,6 +1188,19 @@ def visit(visitor, obj, attr, value):
...
+def visit(visitor, obj):
+    for attr in ("FeatUILabelNameID", "FeatUITooltipTextNameID", "SampleTextNameID"):
+        value = getattr(obj, attr)
+        visitor.seen.add(value)
+    # also include the sequence of UI strings for individual variants, if any
+    if obj.FirstParamUILabelNameID == 0 or obj.NumNamedParameters == 0:
+        return
+    last_name_id = obj.FirstParamUILabelNameID + obj.NumNamedParameters - 1
+    if last_name_id >= 256 and last_namd_id <= 0x7FFF:
+        visitor.seen.update(range(obj.FirstParamUILabelNameID, last_name_id + 1))
```
Comment (khaledhosny) [inline Lib/fontTools/ttLib/tables/_n_a_m_e.py:1201]: Is this check necessary, from the [spec](https://learn.microsoft.com/en-us/typography/opentype/spec/features_ae#tag-cv01--cv99) (emphasis mine):
> The values of featUiLabelNameId, featUiTooltipTextNameId and firstParamUiLabelNameId are **expected** to be in the font-specific name ID range (256–32767), though that is **not a requirement** in this Feature Parameters specification. The value of firstParamUiLabelNameId + numNamedParameters - 1 should not exceed 32767.

Comment (anthrotype) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py:1201]: > Is this check necessary 

which check exactly? `last_name_id >= 256` ?

Comment (behdad) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py:1201]: I agree with Khaled here. No need to check these things...

Comment (khaledhosny) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py:1201]: The whole check. Do we check elsewhere that name IDs are less than 0x7FFF before using them?

Comment (anthrotype) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py:1201]: ok.. I simply copied the code verbatim from hb

Comment (anthrotype) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py:1201]: and there's even a typo! 🤣 

`last_namd_id <= 0x7FFF`

Comment (khaledhosny) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py:1201]: We should drop the check from HarfBuzz then.

**Comment by khaledhosny** [comment]
The issue to be fixed by this should be https://github.com/fonttools/fonttools/issues/3616 not https://github.com/fonttools/fonttools/issues/3516

**Comment by anthrotype** [comment]
thank you Khaled for double checking!

File: Lib/fontTools/ttLib/tables/_n_a_m_e.py (hunk 1 above)
Comment (anthrotype) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py]: I have removed the check. Ideally we'd have some test font to exercise this with. I know I could make one ad-hoc, but do you happen to know a font that contains these which I simply use?

Comment (khaledhosny) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py]: No, I don’t have or know such a font.

Comment (behdad) [reply Lib/fontTools/ttLib/tables/_n_a_m_e.py]: @qxliu76 Can you remove this check from HB too? We don't enforce these kinds of things unless necessary.

**Pull request merged**